x-pack/plugins/security_solution/public/detection_engine/rule_management/components/rule_details/per_field_diff/get_field_diffs_for_grouped_fields.ts

```diff
@@ -24,8 +27,11 @@ export const sortAndStringifyJson = (fieldValue: unknown): string => {
 };
 
 export const getFieldDiffsForDataSource = (
-  dataSourceThreeWayDiff: AllFieldsDiff['data_source']
+  dataSourceThreeWayDiff: RuleFieldsDiffWithDataSource['data_source']
 ): FieldDiff[] => {
+  if (!dataSourceThreeWayDiff) {
+    return [];
+  }
```

Comment:
I guess something is wrong with the types if we need to do this null check? I see type casts in another file that don't look right:

```ts
  switch (fieldName) {
    case 'data_source':
      const dataSourceThreeWayDiff = (fields as RuleFieldsDiffWithDataSource)[fieldName];
      return {
        shouldShowSubtitles: true,
        fieldDiffs: getFieldDiffsForDataSource(dataSourceThreeWayDiff),
      };
```

Reply:
Yep, with your code changes the types are now fixed. Deleted all of these checks. Thanks!